### ISSUE TITLE
adding nodes to an external document with SQF

### ISSUE THREAD

**Issue by rvdb** [schematron-quickfix/sqf#6]
Is there a way to have a Schematron Quick Fix add content to an external document?

I'm using SQF to check the values of some attributes against an authority file containing all distinct values encountered in a collection of XML files (this is just a dynamic editing aid, so there's no way to prescribe legal values in a schema; hence my use of Schematron/SQF). This works well as a lookup tool: if attribute `@x` in document y contains a value that is not listed as a legal value for `@x` in the authority file, an SQF is offered to replace the value of `@x` in document y with one of the legal values found in the authority file.

So far, so good! Yet, I would like to offer the editors an additional option to create a new entry in the authority file for this new value of `@x` in document y. In SQF terms, I would like to "add" an XML element to the authority file containing the new value for `@x`. This way, I could provide the editors with a mechanism to update the authority file as they edit new files. 

I've tried to implement this in a SQF as follows:

```
  <sqf:fix id="test" use-when="@key[normalize-space()]">
    <sqf:description>
      <sqf:title>
        Copy new value for @key (<value-of select="@key"/>) to the authority file.
      </sqf:title>
    </sqf:description>
    <sqf:add match="(doc('authority.xml')//div)[1]">
      <entry name="key" value="@key"/>
    </sqf:add>
  </sqf:fix>
```

But this creates an &lt;entry/> element in document y, instead of authority.xml. 

Am I doing something wrong, or is the `@match` attribute limited to the currently edited document? In other words, does `@match` support selecting external documents using the doc() function?

**Comment by PStellmann** [comment]
Occording to http://www.oxygenxml.com/forum/topic11575.html the modification of other documets by Quick Fixes is scheduled to the next oXygen release.

**Comment by rvdb** [comment]
Thanks, that's good to know.

**Comment by octavianN** [comment]
There is already an issue added for this on the Schematron Quick Fix specification: SQF #5
We implemented this support in oXygen, it will be available in the next oXygen release.

**Comment by rvdb** [comment]
Ah sorry, I should have checked the "closed" issues, my bad! Feel free to delete this non-issue, then.  Great to know it's already implemented and staged for the new Oxygen release; can't hardly wait to see it in action.
